src/dtx/dtx_common.c

```diff
@@ -1812,6 +1820,19 @@ dtx_cont_register(struct ds_cont_child *cont)
 	if (dbca == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
 
+	rc = crt_context_get_timeout(dmi->dmi_ctx, &timeout);
```

Comment:
I am not sure this will give you the right behavior today as the timeout returned will be 0. I grepped through daos code and no one calls crt_context_set_timeout().  Cart on its own leaves it as 0, i've verified it with my local app:

rpc  ERR  src/cart/crt_context.c:1634 crt_context_idx() ALEXMOD: the context timeout is 0

if you are interested in the global timeout used it will be stored in crt_gdata::cg_timeout ("src/cart/crt_internal_types.h")

Reply:
Sure, I will update the patch.

Reply:
Thanks:) cart changes lgtm